dspace-api/src/main/resources/org/dspace/storage/rdbms/sqlmigration/workflow/oracle/xmlworkflow/V6.0_2015.08.11__DS-2701_Xml_Workflow_Migration.sql

```diff
@@ -18,11 +18,65 @@ UPDATE collection SET workflow_step_3 = null;
 ALTER TABLE cwf_workflowitem RENAME COLUMN item_id to item_legacy_id;
 ALTER TABLE cwf_workflowitem ADD item_id RAW(16) REFERENCES Item(uuid);
 UPDATE cwf_workflowitem SET item_id = (SELECT item.uuid FROM item WHERE cwf_workflowitem.item_legacy_id = item.item_id);
+DECLARE
+  COUNT_INDEXES INTEGER;
+  INDEX_NAME VARCHAR(256);
+  TABLE_NAME_D VARCHAR(256) := 'cwf_workflowitem';
+  COLUMN_NAME_D VARCHAR(256) := 'item_legacy_id';
+BEGIN
+  select count(c.INDEX_NAME) into COUNT_INDEXES
+  from USER_INDEXES i, USER_IND_COLUMNS c
+  where i.TABLE_NAME = upper(TABLE_NAME_D)
+        and i.UNIQUENESS = 'UNIQUE'
+        and i.TABLE_NAME = c.TABLE_NAME
+        and i.INDEX_NAME = c.INDEX_NAME
+        and c.COLUMN_NAME = upper(COLUMN_NAME_D);
+
+  IF COUNT_INDEXES > 0 THEN
+    select c.INDEX_NAME into INDEX_NAME
+    from USER_INDEXES i, USER_IND_COLUMNS c
+    where i.TABLE_NAME = upper(TABLE_NAME_D)
+          and i.UNIQUENESS = 'UNIQUE'
+          and i.TABLE_NAME = c.TABLE_NAME
+          and i.INDEX_NAME = c.INDEX_NAME
+          and c.COLUMN_NAME = upper(COLUMN_NAME_D);
+    EXECUTE IMMEDIATE 'ALTER TABLE ' || TABLE_NAME_D || ' DROP constraint ' || INDEX_NAME;
+  END IF;
+END;
+/
```

Comment:
It seems like we should be able to DROP this existing constraint in a separate script that runs either just before (e.g.  named `V5.9_[date]__DS-3829.sql`) or just after this script.  It seems like the goal of this is to simply ensure the constraint is dropped prior to running other updates in the 6.0 script.  In that case, separating this out into a new script whose role is to drop the constraint seems a better option.

You can see other examples of this in other migration scripts...e.g.
https://github.com/DSpace/DSpace/blob/master/dspace-api/src/main/java/org/dspace/storage/rdbms/migration/V1_5_9__Drop_constraint_for_DSpace_1_6_schema.java
(This script drops constraints that cause issues with the 1.5 -> 1.6 database updates...it's named "V1_5_9" so that it is triggered after 1.5 updates but *before* 1.6 updates)

(Also note that we have tools already in place to allow us to find and drop constraints via Java code.  We may be able to use those *instead* of writing complex SQL)